### PR TITLE
feat(MPTv2): Resolve template source as Artifact.

### PR DESCRIPTION
**Pull request by jtk54**

### CODE REVIEW

File: orca-pipelinetemplate/src/test/groovy/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/V1SchemaIntegrationSpec.groovy [hunk 1]
```diff
@@ -48,7 +46,8 @@ class V1SchemaIntegrationSpec extends Specification {
   ObjectMapper objectMapper = new ObjectMapper()
 
   TemplateLoader templateLoader = new TemplateLoader([new ResourceSchemeLoader("/integration/v1schema", objectMapper)])
-  V2TemplateLoader v2TemplateLoader = new V2TemplateLoader([new V2FileTemplateSchemeLoader(objectMapper)])
+  // TODO(jacobkiefer): Fix.
```
Comment (jtk54) [inline orca-pipelinetemplate/src/test/groovy/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/V1SchemaIntegrationSpec.groovy]: Ha, let me fix the tests.

Comment (jtk54) [reply orca-pipelinetemplate/src/test/groovy/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/V1SchemaIntegrationSpec.groovy]: Done.

File: orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/loader/v2/V2TemplateLoader.java [hunk 2]
```diff
@@ -16,48 +16,38 @@
...
+  private OortService oortService;
+
+  private ObjectMapper objectMapper;
 
-  /**
-   * @return a LIFO list of pipeline templates
-   */
-  public V2PipelineTemplate load(TemplateConfiguration.TemplateSource source) {
-    return load(source.getSource());
+  public V2TemplateLoader() {}
```
Comment (ezimanyi) [inline orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/loader/v2/V2TemplateLoader.java]: Why do we need the no-args constructor?  It looks like the resulting `V2TemplateLoader` would just throw NPEs.  Along the same lines, could we set the `oortService` and `objectMapper` to `final`?

Comment (jtk54) [reply orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/loader/v2/V2TemplateLoader.java]: Removed the noargs constructor. I don't think the `final` is necessary.

**Comment by jtk54** [comment]
Comments addressed, PTAL.

**Comment by ezimanyi** [comment]
LGTM

**Pull request merged**